pkg/kafka/kafka-publisher.go

```diff
@@ -260,34 +264,38 @@ func ensureTopic(br *sarama.Broker, timeout time.Duration, topicName string) err
 	}
 }
 
-func waitForBrokerConnection(br *sarama.Broker, config *sarama.Config, timeout time.Duration) error {
+func waitForControllerBrokerConnection(ca *sarama.ClusterAdmin, config *sarama.Config, timeout time.Duration) (cb *sarama.Broker, error) {
 	ticker := time.NewTicker(10 * time.Second)
 	tout := time.NewTimer(timeout)
 	defer func() {
 		ticker.Stop()
 		tout.Stop()
 	}()
+	cb, err := ca.Controller()
```

Comment:
See previous comment, is ca is nil, you will get panic here.

Reply:
fixed in https://github.com/sbezverk/gobmp/pull/236/commits/3c7fbf3dbe99ec0ef3981dcc8634c1931cb1c054